sip-ri/src/main/java/gov/nist/javax/sip/header/extensions/TargetDialog.java

```diff
@@ -0,0 +1,201 @@
+   */
+  public void setCallId(String callId) throws ParseException {
+    try {
+      // Instantiating a new CallIdentifier object using the provided callId
+      this.callId = new CallIdentifier(callId);
+    } catch (Exception e) {
+      // If an exception occurs during instantiation, throw a ParseException
+      throw new ParseException(e.getMessage(), 0);
+    }
+  }
+
+  /**
+   * Gets the Call-Id value of the Target-Dialog header.
+   *
+   * @return the Call-Id value
+   */
+  public String getCallId() {
+    return callId != null ? callId.encode() : null;
+  }
+
+  // New methods for local and remote tags
+
+  /**
+   * Gets the local tag of the Target-Dialog header.
+   *
+   * @return the local tag value
+   */
+  public String getLocalTag() {
+    return localTag;
+  }
+
+  /**
+   * Sets the local tag of the Target-Dialog header.
+   *
+   * @param localTag the local tag value to set
+   * @throws ParseException if the provided localTag is invalid
+   */
+  public void setLocalTag(String localTag) throws ParseException {
+    if (localTag == null || localTag.trim().isEmpty()) {
+      throw new ParseException("Local tag cannot be null or empty", 0);
+    }
+    this.localTag = localTag.trim();
+  }
+
+  /**
+   * Gets the remote tag of the Target-Dialog header.
+   *
+   * @return the remote tag value
+   */
+  public String getRemoteTag() {
+    return remoteTag;
+  }
+
+  /**
+   * Sets the remote tag of the Target-Dialog header.
+   *
+   * @param remoteTag the remote tag value to set
+   * @throws ParseException if the provided remoteTag is invalid
+   */
+  public void setRemoteTag(String remoteTag) throws ParseException {
+    if (remoteTag == null || remoteTag.trim().isEmpty()) {
+      throw new ParseException("Remote tag cannot be null or empty", 0);
+    }
+    this.remoteTag = remoteTag.trim();
+  }
+
+  @Override
+  public StringBuilder encodeBody(StringBuilder retval) {
+    if (callId != null) {
+      retval.append(callId.encode());
+      if (localTag != null) {
+        retval.append(";"); // Use semicolon directly
+        retval.append(localTag);
+      }
+      if (remoteTag != null) {
+        retval.append(";"); // Use semicolon directly
+        retval.append(remoteTag);
+      }
+      // Encode parameters (if any)
+      if (!parameters.isEmpty()) {
+        retval.append(SEMICOLON).append(parameters.encode());
+      }
+    }
+    return retval;
+  }
+
+  @Override
+  public Object clone() {
+    TargetDialog retval = (TargetDialog) super.clone();
+    if (callId != null) {
+      retval.callId = (CallIdentifier) callId.clone();
+    }
+    retval.localTag = localTag;
+    retval.remoteTag = remoteTag;
+    return retval;
+  }
+
+  @Override
+  public void setValue(String value) throws ParseException {
+    // Not implemented yet, throw an exception to indicate it
+    throw new UnsupportedOperationException("setValue method is not implemented yet");
```

Comment:
shouldn't we implement it ?

Reply:
I think this method is odd after close investigation. Since the target dialog header consists of multiple part. Setting the entire value with one string would require parsing it into these individual components which can be error-prone. So as I see, I will just remove this method and leave the setter for tags and callId.

Reply:
fixed it